src/common/strtol.h

```diff
@@ -48,4 +48,20 @@ Target strict_si_cast(const char *str, std::string *err) {
 template <>
 uint64_t strict_si_cast(const char *str, std::string *err);
 
+template<typename T, const unsigned base = 10, const unsigned width = 1>
```

Comment:
In my personal opinion this template needs improvement on interface:
it nicely specifies width of conversion, but *buf must point to beginning of converted output, not end. Otherwise every programmer will make mistake.